src/main/java/com/iota/iri/storage/rocksDB/RocksDBPersistenceProvider.java

```diff
@@ -530,4 +548,54 @@ private void initClassTreeMap(List<ColumnFamilyDescriptor> columnFamilyDescripto
         classTreeMap = MapUtils.unmodifiableMap(classMap);
     }
 
+    private DBOptions createOptions(String logPath, String configFile) throws IOException {
+        DBOptions options = null;
+        File pathToLogDir = Paths.get(logPath).toFile();
+        if (!pathToLogDir.exists() || !pathToLogDir.isDirectory()) {
+            boolean success = pathToLogDir.mkdir();
+            if (!success) {
+                log.warn("Unable to make directory: {}", pathToLogDir);
+            }
+        }
+
+        int numThreads = Math.max(1, Runtime.getRuntime().availableProcessors() / 2);
+        RocksEnv.getDefault()
+            .setBackgroundThreads(numThreads, Priority.HIGH)
+            .setBackgroundThreads(numThreads, Priority.LOW);
+
+        if (configFile != null) {
+            File config = Paths.get(configFile).toFile();
+            if (config.exists() && config.isFile() && config.canRead()) {
+                Properties configProperties = new Properties();
+                
+                try (InputStream stream = new FileInputStream(config)){
+                    configProperties.load(stream);
+                    options = DBOptions.getDBOptionsFromProps(configProperties);
+                } catch (IllegalArgumentException e) {
+                    log.warn("RocksDB configuration file is empty, falling back to default values");
+                }
+            }
+        }
+        if (options == null) {
+            options = new DBOptions()
+                .setCreateIfMissing(true)
+                .setCreateMissingColumnFamilies(true)
+                .setMaxLogFileSize(SizeUnit.MB)
+                .setMaxManifestFileSize(SizeUnit.MB)
+                .setMaxOpenFiles(10000)
+                .setMaxBackgroundCompactions(1)
+                .setAllowConcurrentMemtableWrite(true)
+                .setMaxSubcompactions(Runtime.getRuntime().availableProcessors());
+        }
+        
+        if (!BaseIotaConfig.Defaults.DB_LOG_PATH.equals(logPath) && logPath != null) {
+            if (!options.dbLogDir().equals("")) {
+                log.warn("Defined a db log path in config and commandline; Using the command line setting."); 
+            }
+            
+            options.setDbLogDir(logPath);
+        }
```

Comment:
This is a bit late, I just noticed this code while merging conflicts...
Why is this here @kwek20 ?

Reply:
This is here due to the fact we can define the db log on the config file and the logbath on iri.ini/cmdline. In order to warwn the user they defined it twice, we add a log